### DISCUSSION TITLE
Issue dumping SNES - Street Fighter II Turbo

### DISCUSSION THREAD

**Discussion by vphoebe** [sanni/cartreader#525] [Need help]
## Environment
- hardware V3-ALTER
- software v9.5

## Issue
I have been dumping my SNES cartridges and haven't run into any issues until Street Fighter II Turbo. The cartridge is detected, and the checksum matches, but the CRC32 does not. I have cleaned everything and tried this about 10 times now. Same thing and the same CRC32 result every time. Dump logs: 
```
OSCR HW3 V9.5


Searching database...
Found

Name: StreetFighter2T
Type: HiROM FastROM
ICs: ROM ONLY
ROM Size: 32Mbit
Banks: 64 Chips: 0
Sram Size: 0Kbit
ROM Version: 1.0
Checksum: 119A
 
 


Saving to SNES/ROM/StreetFighter2T/90/...
Dumping HiRom...
[*******************]
Chksum...OK
CRC32... 503A9A89 -> Not found
```

If I open the file in an emulator to test, it boots correctly. If I compare with a known good dump in HexFiend, I get this result (known good on left vs. my dump on right.) 

So the files appear to be the same up until that offset, but my dump appears to just be 1.5MB larger. I assume that it just begins to "repeat" until the end of the file. I am wondering if this is something I can resolve in the cartridge reader settings, or can handle it in some other way. I didn't have any CRC32 errors with the other 11 cartridges in my small collection. 

I also noticed that on [this site](https://superfamicom.org/info/street-fighter-2-turbo-hyper-fighting), the ROM size is listed at 20Mbit, but the cartridge reader database says `ROM Size: 32Mbit`. If that's the case, potentially the database entry for this game is incorrect, causing the duplication in the dump? Any help or advice is appreciated, thank you!

**Comment by sanni** [comment]
Can you update to the [newest Github files](https://github.com/sanni/cartreader/archive/refs/heads/master.zip) (firmware and database files) and try again? This should be already fixed unless I messed something up. 😄

**Comment by vphoebe** [comment]
That worked perfectly! I saw it change itself to 20Mbit and the dump now has the correct CRC32. Thanks for the response! Was that part of release 9.5 or has that been added/fixed since? I thought I updated to 9.5 while troubleshooting but perhaps I missed a step or something. In any case it works now, so thanks again!

**Comment by keungk** [comment]
i did the above, and its also fixed Super Mario Kart (snes) and Metroid Zero Mission (GBA) (both EUR) as well. The ROM sizes were too big before.

**Reply by RascalDungeon** [reply]
I updated the firmware and sd files and I still have issues with Super Mario Kart. It loads to the Nintendo logo and makes the ding sound, then freezes.

**Reply by RascalDungeon** [reply]
I should add that the SMK dump plays in an emulator. Just not on an Everdrive on original hardware.

**Reply by PsyK0p4T** [reply]
Don't bother to run it on an emulator, the only important thing is your rom hashes.
- What version of SMK are you trying to dump ?
- What size is your current dump ?
- What is the CRC32 of your dump ?
- Please add the logs generated by OSCR

**Reply by RascalDungeon** [reply]
It is the USA regular release. Rev0.
The dump is 524,288 bytes.
The CRC32 is CD80DB86.

Here is the log:


OSCR HW3 V12.3

[+] SNES/SFC (CLK0+1)

[+] SNES/SFC cartridge


Searching database...
Checksum: EB44
Header CRC32: 17B0638D
Found

Title: SUPER MARIO KART
Revision: 0
Type: HiROM FastROM
ICs: ROM DSP1 SAVE
ROM Size: 512 KB (8 banks)
Save Size: 2 KB
Checksum: EB44
Press Button...

[+] Read ROM


Saving to SNES/ROM/SUPER MARIO KART/349/...
Dumping HiRom...
[*******************]
Checksum... EB44 -> OK
CRC32... CD80DB86 -> Super Mario Kart (USA).sfc
Press Button...

**Reply by PsyK0p4T** [reply]
Sorry for the late answer. Your dump looks totally fine then, crc32 CD80DB86 is the expected one and matches previous redumps for that cart. If it doesn't load correctly elsewhere (emulator or flashcart) then it's not the rom fault really.

**Reply by RascalDungeon** [reply]
So the problem is with the Everdrive? I'll ask on the Krikzz forums and see what people there say.

**Reply by PsyK0p4T** [reply]
Could be many things... from your SD card (partitionning, brand, bad copy), to the flashcard (firmware, rom compatibility...). At least your rom is not the cause for me. Try to run it on a PC emulator to confirm.

**Reply by RascalDungeon** [reply]
Found my answer here: 

https://krikzz.com/forum/index.php?topic=3281.msg31872#msg31872

Turns out I needed the DSP1b.bin file for the special chip contained in Mario Kart added to my SD2SNES folder.

**Reply by 7ank0v1c** [reply]
Ah really, that's an interesting issue. Glad you fixed it mate
